src/operators/del-operators.ads

```diff
@@ -2,22 +2,16 @@ package Del.Operators is
    type Linear_T is new Func_T with null record;
    type Linear_Access_T is access all Linear_T'Class;
 
-   overriding function Forward (L : Linear_T; X : Tensor_T) return Tensor_T;
-   overriding function Backward (L : Linear_T; Dy : Tensor_T) return Tensor_T;
+   overriding function Forward (L : in out Linear_T; X : Tensor_T) return Tensor_T;
+   overriding function Backward (L : in Linear_T; Dy : Tensor_T) return Tensor_T;
```

Comment:
We don't need to add the _in_ keyword. (Default parameter uses _in_)

Reply:
I need to use "in out" for the parameter to modify the map @Slamdir